src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,30 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    m_pContext->getSingletonMap()->defineSingleton(child_group->objectName(),
+                                                   child_group);
+    child_group->hide();
+}
+
+QWidget* LegacySkinParser::parseSingletonContainer(QDomElement node) {
```

Comment:
The idea is that the Map is in charge of managing the Containers.  The constructor of the container takes the widget being pointed at, and I really don't want callers to have to think about getting the pointer out of the map and then passing it to the container.  If that was the case, then callers are suddenly responsible for managing their own maps and could get into trouble.  I was even thinking of making the Container constructor still private, and making the Map a friend class so that people couldn't subvert the design and try to construct their own containers.

The idea is to keep the interface too simple to mess up -- tell the map "here is a new singleton," and later "give me the instance of the singleton with this name.